### PR TITLE
Restrict character types in question option Admin IDs

### LINKED ISSUE #5839: Restrict characters allowed in option admin names
Building on #4862, restrict option admin names to a-z, A-Z,0-9, hyphens, and dashes.

Existing option admin names would be unaffected.

The goal is to make it easier to write code that processes these admin names. For example, writing SQL to target `option_1` is much easier then `option "alpha" #1`.

Approved by Exygy, Seattle, Arkansas, and Bloomington.

### PR DESCRIPTION
**Pull request by avaleske**
### Description

Restricts multi-option question admin IDs to letters, numbers, underscores, and dashes.

The bulk of the changes are:
- Updating tests to use valid admin names
- Removing the outdated string from the other language files

The interesting changes are in:
- `admin_validation.ts`
- `QuestionDefinition.java`

## Release notes

Restricts multi-option question admin IDs to letters, numbers, underscores, and dashes. This is so that writing code that processes admin IDs will be easier and less error-prone.

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | bug | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers).
- [x] Created unit and/or browser tests which fail without the change (if possible)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)
- [n/a] Extended the README / documentation, if necessary

#### User visible changes

- [x] Followed steps to [internationalize new strings](https://github.com/civiform/civiform/wiki/Internationalization-%28i18n%29#internationalization-for-application-strings)
- [x] Wrote browser tests using the [validateAccessibility](https://sourcegraph.com/github.com/civiform/civiform/-/blob/browser-test/src/support/index.ts?L437:14&subtree=true) method
- [n/a] Tested on mobile view. See [mobile device mode](https://developer.chrome.com/docs/devtools/device-mode/)
- [n/a] Manually tested at 200% size
- [n/a] Manually evaluated tab order

#### New Features

- [n/a] Add new FeatureFlag env vars to `server/conf/helper/feature-flags.conf`
- [n/a] Conditioned new functionality on a [FeatureFlag](https://github.com/civiform/civiform/wiki/Feature-Flags)
- [n/a] Wrote browser tests with the feature flag off and on, etc.

### Instructions for manual testing

Try adding an invalid admin name and see that it's caught by the validation.

### Issue(s) this completes

Fixes #5839

### CODE REVIEW

**Comment by avaleske** [comment]
Current dependencies on/for this PR:
* main
  * **PR #5929** <a href="https://app.graphite.dev/github/pr/civiform/civiform/5929" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/civiform/civiform/5929?utm_source=stack-comment).

File: browser-test/image_snapshots/question_lifecycle_test/question-with-invalid-admin-names-error.png
Comment (bion) [inline browser-test/image_snapshots/question_lifecycle_test/question-with-invalid-admin-names-error.png:1]: issue (blocking): need to fix layout for the inputs so the validation error messages don't throw the grid off

File: server/app/assets/javascripts/admin_validation.ts [hunk 1]
```diff
@@ -76,6 +76,30 @@ class AdminValidationController {
...
+    for (const inputElement of inputElements) {
+      // Assert that the admin name is not blank, and that it only
+      // contains 0-9, a-z, A-Z, _, and -
+      const inputIsValid = /^[0-9a-zA-Z_-]+$/.test(inputElement.value)
+      this.updateFieldErrorState(inputElement, errorClass, inputIsValid)
+      if (!inputIsValid) {
+        multiOptionIsValid = inputIsValid
+      }
+    }
+    return multiOptionIsValid
```
Comment (bion) [inline server/app/assets/javascripts/admin_validation.ts:100]: suggestion (non-blocking): could do?
```suggestion
    for (const inputElement of inputElements) {
      // Assert that the admin name is not blank, and that it only
      // contains 0-9, a-z, A-Z, _, and -
      const inputIsValid = /^[0-9a-zA-Z_-]+$/.test(inputElement.value)
      this.updateFieldErrorState(inputElement, errorClass, inputIsValid)
      
      if (!inputIsValid) {
        return false
      }
    }
    
    return true
```

**Review by bion — CHANGES_REQUESTED** [review]

File: browser-test/image_snapshots/question_lifecycle_test/question-with-invalid-admin-names-error.png
Comment (avaleske) [reply browser-test/image_snapshots/question_lifecycle_test/question-with-invalid-admin-names-error.png:1]: That's captured by #5612, would you like me to shelve this change and do that redesign first?

File: server/app/assets/javascripts/admin_validation.ts (hunk 1 above)
Comment (avaleske) [reply server/app/assets/javascripts/admin_validation.ts:100]: We need to visit every option admin name to decide if its error message should be shown. If we exit on the first failure, we won't visit them all.

Comment (bion) [reply server/app/assets/javascripts/admin_validation.ts:100]: oh, woops. sorry!

File: browser-test/image_snapshots/question_lifecycle_test/question-with-invalid-admin-names-error.png
Comment (bion) [reply browser-test/image_snapshots/question_lifecycle_test/question-with-invalid-admin-names-error.png:1]: nope, got it

**Review by bion — APPROVED** [review]

**Pull request merged**